ddtrace/pin.py

```diff
@@ -104,25 +117,33 @@ def onto(self, obj, send=True):
                 return obj.__setddpin__(self)
 
             pin_name = _DD_PIN_PROXY_NAME if isinstance(obj, wrapt.ObjectProxy) else _DD_PIN_NAME
+
+            # set the target reference; any get_from, clones and retarget the new PIN
+            self._target = obj
```

Comment:
Everytime a `Pin` is attached `onto` an object, we set the pin's owner.

Reply:
Any change it can badly affect the garbage collector?

Reply:
Yes I'm investigating if it's better to just use the memory address instead of the object pointer.

Reply:
if we use `id(obj)` we avoid any kind of reference counting for the GC. This introduces a little overhead:
```
In [53]: timeit.timeit("id({'a':5})", number=10000)
Out[53]: 0.0017991065979003906

In [34]: timeit.timeit("{'a':5}", number=10000)
Out[34]: 0.0015370845794677734
```

that I think it's not relevant. The only caveat is that we have these results for CPython, and it could be different for other interpreters, but at the moment we officially support only CPython.